ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala

```diff
@@ -95,12 +95,39 @@ trait WriteService
     * @param estimatedInterpretationCost Estimated cost of interpretation that may be used for
     *                                    handling submitted transactions differently.
     * @param telemetryContext            Implicit context for tracing.
-    * @return an async result of a SubmissionResult
     */
   def submitTransaction(
       submitterInfo: SubmitterInfo,
       transactionMeta: TransactionMeta,
       transaction: SubmittedTransaction,
       estimatedInterpretationCost: Long,
```

Comment:
@andreaslochbihler-da : KV implementations will need to check the requested `SubmitterInfo.deduplicationPeriod` against the `maxDeduplicationPeriod` provided in the `LedgerConfiguration`. 

Would you expect that to be provided by the caller or read on-demand by the KV implementations?

Reply:
The KV implementations anyway need some hook into persistence (on the indexer) for filling in completion offsets, so they can get the current ledger configuration from there, possibly with an in-memory cache to avoid frequent reads from the DB. Other Read/WriteService implementations may not need the current `LedgerConfiguration`, so it would feel strange to add such a parameter to the Read/WriteService.

Reply:
I thought we'd delegate the filling of completion offsets to the indexer due to the complex interactions with parallel, batched ingestion.

I agree though that we should have clear rules of what is passed as a parameter on a `WriteService` method. I'd suggest: the Ledger API server passes explicit immutable references for all mutable data read from the IndexDB.

This way `WriteService` implementations can use the same version of that data that was used by the Ledger API server.

So in this case, we should check whether the Ledger API server uses the `LedgerConfiguration` during interpretation or not. If it does, it should pass it on. If not, the implementation should look it up.

Reply:
@andreaslochbihler-da : as per our discussion, let's add the `LedgerConfiguration` that was used to interpret the command and determine it's ledger effective time.

Reply:
Added a `SubmissionContext` parameter for such context information.